internal/reconciler/metricpipeline/status.go

```diff
@@ -77,6 +85,49 @@ func (r *Reconciler) updateStatus(ctx context.Context, pipelineName string, lock
 	return setCondition(ctx, r.Client, &pipeline, pending)
 }
 
+func (r *Reconciler) updateAgentStatus(ctx context.Context, pipelineName string) error {
+	log := logf.FromContext(ctx)
+
+	var pipeline telemetryv1alpha1.MetricPipeline
+	if err := r.Get(ctx, types.NamespacedName{Name: pipelineName}, &pipeline); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+
+		return fmt.Errorf("failed to get MetricPipeline: %v", err)
+	}
+
+	agentEnabled := isMetricAgentRequired(&pipeline)
+
+	if agentEnabled {
```

Comment:
Let's avoid deep nesting:
```suggestion
	if !agentEnabled {
	   return nil
	}
```